src/MonitoringFunctions/DataService/Kusto/DirectJsonMappingResolver.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Kusto.Data.Common;
+using Newtonsoft.Json.Serialization;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace MonitoringFunctions.DataService.Kusto
+{
+    /// <summary>
+    /// Creates column mapping for objects whose json property names directly match the column name in the corresponding Kusto table.
+    /// </summary>
+    internal sealed class DirectJsonMappingResolver : IJsonColumnMappingResolver
+    {
+        public IEnumerable<ColumnMapping> GetColumnMappings<T>() where T : IKustoTableRow
+        {
+            DefaultContractResolver contractResolver = new DefaultContractResolver();
+            var contract = contractResolver.ResolveContract(typeof(T)) as JsonObjectContract;
+
+            if (contract == null)
+            {
+                throw new Exception($"Failed to resolve contract. Automatic column mapping is not possible with this type {typeof(T)}.");
```

Comment:
minor: is there a more relevant Exception type for this case?

Reply:
There isn't a TypeArgumentException type in BCL, so I settled for ArgumentException.